src/python/pants/init/extension_loader.py

```diff
@@ -136,17 +137,29 @@ def load_backend(build_configuration: BuildConfiguration.Builder, backend_packag
     :raises: :class:``pants.base.exceptions.BuildConfigurationError`` if there is a problem loading
       the build configuration.
     """
+    backend_package = backend_package.strip()
+    is_templated_backend = ('(' in backend_package) and backend_package.endswith(')')
+    if is_templated_backend:
+        kwargs_str = backend_package[backend_package.find('('):]
+        backend_package = backend_package[:backend_package.find('(')]
```

Comment:
Nifty idea. I'd suggest setting `is_templated_backend` in the presence of `(` alone, and err unless it ends with `)` to help catch typos etc.
Also, is_templated_backend could be the index right away, reducing the search for the `(` from three times to one.